userspace/libsinsp/event.cpp

```diff
@@ -2007,6 +2038,55 @@ const char* sinsp_evt::get_param_as_str(uint32_t id, OUT const char** resolved_s
 							  flags->name);
 			}
 
+			break;
+		}
+	case PT_MODE:
+		{
+			uint32_t val = *(uint32_t *)payload;
+			SET_NUMERIC_FORMAT(prfmt, param_fmt, PRIo32, PRId32, PRIX32);
+			snprintf(&m_paramstr_storage[0],
+					m_paramstr_storage.size(),
+					prfmt, val);
+
+			const struct ppm_name_value *mode = (const struct ppm_name_value *)m_info->params[id].info;
+			const char *separator = "";
+			uint32_t initial_val = val;
+			uint32_t j = 0;
+
+			while(mode != NULL && mode->name != NULL && mode->value != initial_val)
+			{
+				// If mode is 0, then initial_val needs to be 0 for the mode to be resolved
+				if((mode->value == 0 && initial_val == 0) ||
+				   (mode->value != 0 && (val & mode->value) == mode->value && val != 0))
+				{
+					if(m_resolved_paramstr_storage.size() < j + strlen(separator) + strlen(mode->name))
+					{
+						m_resolved_paramstr_storage.resize(m_resolved_paramstr_storage.size() * 2);
```

Comment:
This isn't guaranteed to resize the vector to big enough, is it? Shouldn't this resize be the exact additional bytes needed or the if two lines up should be a while?

Reply:
the vector is big enough because the bitmask we have can expand maximum to 96 when all modes are used, however I'm pushing a change to make it match the exact size so that we don't oversize that vector just for this.

Reply:
Also, that is how I'm using mode right now but there's no guarantee that one might not use it with a different set of parameters so this proposed change makes a lot of sense in order to be defensive.